### ISSUE TITLE
[DOCUMENTATION] "colour.SpectralDistribution" Cubic Spline Interpolator produces unexpected results.

### ISSUE THREAD

**Issue by sliedes** [colour-science/colour#997]
### Description

Ok, so I don't know if this is a well known pitfall; I think at least a warning in the docs might be in order.

I have some spectral data that I extracted from a PDF datasheet for a LED using some black magic (it was there in vector form). The data is non-uniformly spaced, consists of values between 0.0 and 1.0, and as far as I can tell is in no way clearly pathological. The spectrum looks like this:

![deep_red](https://user-images.githubusercontent.com/2307997/172490236-6cf34031-20d9-4196-8d77-7d4bbcbff802.png)

For this data, using the default cubic spline interpolator in `SpectralDistribution`, `sd_to_XYZ` returns (3.917, 0.186, -6.172), which is clearly bogus because the XYZ coordinates should be nonnegative.

With linear interpolator, it instead returns a more reasonable value of (5.423, 2.110, 0.004).

### Code for Reproduction

```python
import colour
import numpy as np

sd_values = np.array(
    [
        [3.60000000e02, 0.00000000e00],
        [3.60998118e02, 3.55361453e-03],
        [3.61996237e02, 0.00000000e00],
        [3.67981852e02, 0.00000000e00],
        [3.68979971e02, 1.23663947e-03],
        [3.69978089e02, 0.00000000e00],
        [5.96462764e02, 1.14929318e-04],
        [5.99455572e02, 1.00218366e-03],
        [6.01451809e02, 2.19744857e-03],
        [6.03446498e02, 3.87081945e-03],
        [6.05441187e02, 6.27973796e-03],
        [6.07437424e02, 9.24491438e-03],
        [6.09432114e02, 1.27893346e-02],
        [6.10430232e02, 1.43799563e-02],
        [6.12426469e02, 1.86829100e-02],
        [6.14421158e02, 2.32800827e-02],
        [6.17415513e02, 3.19319618e-02],
        [6.18412084e02, 3.59912654e-02],
        [6.19410203e02, 3.93334100e-02],
        [6.21406440e02, 4.72773244e-02],
        [6.22403011e02, 5.21273417e-02],
        [6.23401129e02, 5.59613837e-02],
        [6.26395484e02, 7.35455695e-02],
        [6.27392055e02, 8.01241237e-02],
        [6.31382981e02, 1.12143432e-01],
        [6.33377671e02, 1.33111137e-01],
        [6.37370144e02, 1.86383174e-01],
        [6.40362952e02, 2.40183887e-01],
        [6.42357642e02, 2.81195265e-01],
        [6.46350115e02, 3.88465694e-01],
        [6.52335731e02, 5.85803931e-01],
        [6.53332302e02, 6.19965521e-01],
        [6.54330420e02, 6.58710493e-01],
        [6.55328539e02, 7.03799563e-01],
        [6.56326657e02, 7.42811171e-01],
        [6.57324775e02, 7.90594185e-01],
        [6.62312272e02, 9.81519366e-01],
        [6.63310391e02, 1.00000000e00],
        [6.64308509e02, 1.00000000e00],
        [6.65306628e02, 9.97572693e-01],
        [6.66303199e02, 9.68909321e-01],
        [6.67301317e02, 9.16147569e-01],
        [6.68299436e02, 8.38846110e-01],
        [6.70295672e02, 6.55832663e-01],
        [6.71292243e02, 5.56667050e-01],
        [6.72290362e02, 4.69913803e-01],
        [6.73286933e02, 3.93067464e-01],
        [6.74285051e02, 3.27189978e-01],
        [6.75283170e02, 2.71256177e-01],
        [6.76281288e02, 2.27445121e-01],
        [6.77277859e02, 1.89991955e-01],
        [6.79274096e02, 1.33800713e-01],
        [6.81270333e02, 9.53867372e-02],
        [6.82266903e02, 8.01700954e-02],
        [6.83265022e02, 6.83185841e-02],
        [6.85261259e02, 4.99574762e-02],
        [6.87255948e02, 3.65751063e-02],
        [6.88254067e02, 3.07964602e-02],
        [6.89252185e02, 2.59004712e-02],
        [6.90250303e02, 2.30778071e-02],
        [6.91246874e02, 1.97494541e-02],
        [6.94241230e02, 1.28123204e-02],
        [6.98230608e02, 7.28192162e-03],
        [7.01224964e02, 4.96494656e-03],
        [7.02221534e02, 4.67072750e-03],
        [7.03219653e02, 3.73290426e-03],
        [7.04217771e02, 3.53522584e-03],
        [7.06212461e02, 2.55602804e-03],
        [7.10203387e02, 1.48028962e-03],
        [7.11201505e02, 1.83886910e-03],
        [7.12199624e02, 1.25502816e-03],
        [7.19183358e02, 5.28674865e-04],
        [7.21178047e02, 6.43604183e-04],
        [7.23174284e02, 2.66636019e-04],
        [7.29161447e02, 2.57441673e-04],
        [7.30158018e02, 0.00000000e00],
        [7.32154255e02, 2.89621883e-04],
        [7.38141418e02, 4.13745546e-05],
        [7.40136107e02, 4.87300310e-04],
        [7.42130796e02, 1.97678428e-04],
        [7.43128915e02, 3.95356856e-04],
        [7.44127033e02, 2.75830364e-05],
        [7.53107004e02, 0.00000000e00],
        [7.54103575e02, 5.05689001e-04],
        [7.55101693e02, 0.00000000e00],
        [7.59092620e02, 4.91897483e-04],
        [7.60090738e02, 0.00000000e00],
        [7.63083546e02, 5.14883347e-04],
        [7.64081664e02, 5.97632456e-04],
        [7.65078235e02, 1.05734973e-04],
        [7.82041606e02, 9.65406275e-05],
        [7.84036295e02, 1.19526491e-04],
        [7.85032866e02, 9.33226066e-04],
        [7.86030985e02, 1.10332146e-04],
        [7.89025340e02, 0.00000000e00],
        [8.00000000e02, 0.00000000e00],
    ]
)

sd1 = colour.SpectralDistribution(dict(sd_values))
print(tuple(colour.sd_to_XYZ(sd1, method="Integration")))
# Output: (3.9168753204573155, 0.18571268240738295, -6.1720721214304559)

sd2 = colour.SpectralDistribution(dict(sd_values), interpolator=colour.LinearInterpolator)
print(tuple(colour.sd_to_XYZ(sd2, method="Integration")))
# Output: (5.4238696472564092, 2.110494312330923, 0.0044943059334115326)
```

### Environment Information

```shell
===============================================================================
*                                                                             *
*   Interpreter :                                                             *
*       python : 3.10.4 (main, May 14 2022, 05:21:19) [GCC 12.1.0]            *
*                                                                             *
*   colour-science.org :                                                      *
*       colour : 0.4.1                                                        *
*                                                                             *
*   Runtime :                                                                 *
*       imageio : 2.16.1                                                      *
*       matplotlib : 3.5.1                                                    *
*       networkx : 2.8                                                        *
*       numpy : 1.22.3                                                        *
*       pandas : 1.4.2                                                        *
*       scipy : 1.8.0                                                         *
*       sklearn : 1.0.2                                                       *
*       tqdm : 4.64.0                                                         *
*                                                                             *
===============================================================================
defaultdict(dict,
            {'Interpreter': {'python': '3.10.4 (main, May 14 2022, 05:21:19) [GCC 12.1.0]'},
             'colour-science.org': {'colour': '0.4.1'},
             'Runtime': {'imageio': '2.16.1',
              'matplotlib': '3.5.1',
              'networkx': '2.8',
              'numpy': '1.22.3',
              'pandas': '1.4.2',
              'scipy': '1.8.0',
              'sklearn': '1.0.2',
              'tqdm': '4.64.0'}})
```

**Comment by sliedes** [comment]
> I would argue that the data is _extremely_ pathological, the non-uniformity is dramatic in some areas, I have actually never seen anything like that. Did you use [WebPlotDigitizer](https://automeris.io/WebPlotDigitizer/)?

Ah, you are actually right; I missed that because I only looked at the plot. Your CubicSplineInterpolator result plots explain a lot :)

I actually didn't use WebPlotDigitizer, but thanks for the link, it looks very useful and I've been looking for something like that! What I did was a bit more involved; I decompressed a PDF datasheet and interpreted the coordinates in it used to plot the graph.

I agree that a note in documentation especially about extreme non-uniformity would have been quite helpful here. Then I know I always manage to come up with novel pathological cases :D

Thanks!
